loki/transformations/loop_blocking.py

```diff
@@ -0,0 +1,310 @@
+# (C) Copyright 2018- ECMWF.
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+from pymbolic.primitives import Expression
+
+from loki import Subroutine, ir, Assignment, Sum, Product, Loop, LoopRange, IntLiteral, \
+    Quotient, InlineCall, DeferredTypeSymbol, Transformer, Array, Variable, Scalar, \
+    SubstituteExpressions, FindVariables, parse_expr, RangeIndex
+
+
+def ceil_division(iexpr1: Expression, iexpr2: Expression) -> Expression:
+    """
+    Returns ceiled division expression of two integer expressions iexpr1/iexpr2.
+    """
+    return Sum(children=(Quotient(numerator=Sum(children=(iexpr1, IntLiteral(-1))),
+                                  denominator=iexpr2), IntLiteral(1)))
+
+
+def negate(i: Expression) -> Expression:
+    return Product((IntLiteral(-1), i))
+
+
+def num_iterations(loop_range: LoopRange) -> Expression:
+    """
+    Returns total number of iterations of a loop.
+
+    Given a loop, this returns an expression that computes the total number of
+    iterations of the loop, i.e.
+    (start,stop,step) -> ceil(stop-start/step)
+    """
+    start = loop_range.start
+    stop = loop_range.stop
+    step = loop_range.step
+    if step is None:
+        return stop if isinstance(start, IntLiteral) and start.value == 1 else Sum(
+            (stop, negate(start), IntLiteral(1)))
+    return Sum((Quotient(Sum((stop, negate(start))), step), IntLiteral(1)))
+
+
+def normalized_loop_range(loop_range: LoopRange) -> LoopRange:
```

Comment:
I could add it to Loki.expression.symbolic, but I also think that this could be a property of the LoopRange class itself (as you suggested with the num_iterations utility above).